cmd/metal-api/internal/service/v1/machine.go

```diff
@@ -380,6 +435,20 @@ func NewMachineResponse(m *metal.Machine, s *metal.Size, p *metal.Partition, i *
 			MachineNetworks: networks,
 			Succeeded:       m.Allocation.Succeeded,
 		}
+
+		if m.Allocation.Reinstall {
```

Comment:
So... only during reinstallation you have the MachineSetup fields available and when it succeeded this is gone? Aren't these fields true before and after a reinstall?

Reply:
Just make sure that this is only a mapping function. It should be dumb.